pkg/backup/table_pattern.go

```diff
@@ -109,6 +109,31 @@ func getTableListByPatternLocal(metadataPath string, tablePattern string, skipTa
 	return result, nil
 }
 
+func getTableListByRestoreDatabaseMappingRule(originTables *ListOfTables, dbMapRule map[string]string) error {
+	result := ListOfTables{}
+	dbMetaMap := make(map[string]metadata.TableMetadata, 0)
+	for i := 0; i < len(*originTables); i++ {
+		table := (*originTables)[i]
+		dbMetaMap[table.Database] = table
+	}
+	for i := 0; i < len(*originTables); i++ {
+		originDBMeta := (*originTables)[i]
+		if targetDB, ok := dbMapRule[originDBMeta.Database]; ok {
+			originDBMeta.Query = strings.Replace(
+				originDBMeta.Query, originDBMeta.Database+".", targetDB+".", 1,
```

Comment:
we have some corner cases
when 
database name escaped with **`** character
also we shall avoid corner cases when .Query contains something like
```CREATE VIEW db1.view AS SELECT * FROM db2_db1.another_table```

i propose replace database name with regexp something like that
```golang
regExp := `^CREATE (TABLE|VIEW|MATERIALIZED VIEW|DICTiONARY|FUNCTION) ([`]*)(db)([`]*)\.`
toStr := "CREATE $1 $2new_db$4."
```